src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -257,15 +263,24 @@ protected void AddSecrets(TransactionBuildContext context)
             if (this.privateKeyCache.TryGetValue(cacheKey, out SecureString secretValue))
             {
                 privateKey = wallet.Network.CreateBitcoinSecret(secretValue.FromSecureString()).PrivateKey;
-                this.privateKeyCache.Set(cacheKey, secretValue, new TimeSpan(0, 5, 0));
+                TimeSpan duration;
+                if (!this.privateKeyCacheTimeout.TryGetValue(cacheKey, out duration))
```

Comment:
@mikedennis do you know why isn't `AddSecrets()` using `CacheSecret()`?

Reply:
It's already got cachekey so would be a bit repetitive code.  Could probably move the code for updating cache and timeout to a common method and call from both though?

Reply:
What I'm saying is that the method AddSecrets() has code that does what CacheSecret () does but not using it.

this is inside AddSecrets():
https://github.com/stratisproject/StratisBitcoinFullNode/blob/d7d152346200ff65d973626f3b3d17be996e0042/src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs#L245-L262

this is CacheSecrets():
https://github.com/stratisproject/StratisBitcoinFullNode/blob/d7d152346200ff65d973626f3b3d17be996e0042/src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs#L182-L200

If you change CacheSecrets to return the private key you can use this method inside AddSecrets rather than having the same code duplicated.

Reply:
Yes that would work. If I changed CacheSecret() to return the private key then we could be used from AddSecrets().  I'm debating whether we might want to change the name of the method then to GetPrivateKey() or something like that?

Reply:
I think you can leave the name as is.
CacheSecret caches the secret and returns the cached item. I think the caching is the real function here.
A bit like when you create an object via a POST api call and you return the created object because it may be useful. Same here.